### PR TITLE
New Rule: LT15

### LINKED ISSUE #2634: New rule proposal: avoid several empty lines in a row.
I don't know if several empty lines is something that makes sense in some dialects, but if not, I think SQLFluff should have a rule to flag this.

**Anti-pattern**
```sql
SELECT * FROM my_table;


SELECT * FROM my_other_table;
```

**Best practice**
```sql
SELECT * FROM my_table;

SELECT * FROM my_other_table;
```

Removing all empty lines makes no sense at all as it adds clean space to separate a list of statements, like a new paragraph in a novel. But having too much of them in a row creates a too big space that can make your SQL hard to read.

This style rule is also present in [JavaScript](https://eslint.org/docs/rules/no-multiple-empty-lines) for instance, and probably from other languages.

Technically, I would go on looking for `newline` segment type, and if the (at least) 2 next segments type encountered are also `newline`s, the rule should flag. The fix to provide is to simply keep 2 newlines in a row when 3+ are found.

Thanks a lot!

### PR DESCRIPTION
**Pull request by WittierDinosaur**
### Brief summary of the change made
This PR adds a long-requested rule. It allows the control of how many blank lines there can be in a row. It is configurable to different values inside and outside of a statement.

This changes the default rule set - and so should trigger a minor release.

Fixes #2634 

### Are there any other side effects of this change that we should be aware of?
No

### Pull Request checklist
- [x] Please confirm you have completed any of the necessary steps below.

- Included test cases to demonstrate any code changes, which may be one or more of the following:
  - `.yml` rule test cases in `test/fixtures/rules/std_rule_cases`.
  - `.sql`/`.yml` parser test cases in `test/fixtures/dialects` (note YML files can be auto generated with `tox -e generate-fixture-yml`).
  - Full autofix test cases in `test/fixtures/linter/autofix`.
  - Other.
- Added appropriate documentation for the change.
- Created GitHub issues for any relevant followup/future enhancements if appropriate.

### CODE REVIEW

File: test/fixtures/rules/std_rule_cases/LT15.yml [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    ;
+
+    SELECT foo
+    ;
+  configs:
+    rules:
+      layout.newlines:
+        maximum_empty_lines_between_statements: 1
+
+test_fail_bad_edge_case:
```
Comment (WittierDinosaur) [inline test/fixtures/rules/std_rule_cases/LT15.yml:57]: @alanmcruickshank here is the edge case that i'd like to avoid

Comment (alanmcruickshank) [reply test/fixtures/rules/std_rule_cases/LT15.yml:57]: I see what you mean 🤔 . I'm really not sure this is a problem. Personally I feel like the semicolon "belongs" to the preceding statement, but I also think a comma "belongs" to the preceding line. Leaving this open so that we're a bit more lenient on whether the "between statements" gap is _before_ or _after_ the semicolon I think is ok.

I don't think hard coding the behaviour is worth the extra complexity unless we also give people the option to configure it in case they want to do.... _leading semicolons?!_.

Comment (alanmcruickshank) [reply test/fixtures/rules/std_rule_cases/LT15.yml:57]: I think I' trying to say: I think this PR is ok as-is. We can always optionally tighten it later.

Comment (alanmcruickshank) [reply test/fixtures/rules/std_rule_cases/LT15.yml:57]: I'll add a comment to that test case to explain

**Review by alanmcruickshank — APPROVED** [review]
Nice. I don't think the edge case is a problem at the moment. Yeah some people will disagree but I think that's small potatoes and can be handled at a later date. This already adds a bunch of value as-is.
